### PR TITLE
Add pre/post analyze script

### PR DESCRIPTION
**Pull request by tmsblgh**
Analysis container should run the analysis in the workspace prepared by the controller.

- [ ] - extract the zip file
- [ ] - update file paths
- [ ] - run the analysis
- [ ] - update the file paths back to the original in the results
- [ ] - save the analysis output to a file (stdout, stderr, returncode)
- [ ] - compress the results? (maybe the controller should do this if there are multiple analyzer containers for one analysis id)

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: '3.6'  
+services:  
+  analyzer:
+    image: codechecker
+    command: tail -F /dev/null
```
Comment (gyorb) [inline docker-compose.yaml]: why is this needed?

File: remote/analyze_handler.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    ANALYZE_COMPLETED = 'ANALYZE_COMPLETED'
+
+
+def preAnalyze(analyze_id, part_number):
+    LOGGER.info('Pre Analyze Step')
+
+    file_name = 'source'
+    file_extension = '.zip'
+
+    file_path = os.path.join('/workspace', analyze_id)
```
Comment (gyorb) [inline remote/analyze_handler.py]: I think the `workspace` path could be an argument for the whole script.

File: remote/analyze_handler.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+    with open(file_to_analyze_path) as read_in_file:
+        try:
+            read_in_file_path = read_in_file.readline()
+        except Exception:
+            LOGGER.error("Failed to read in build command.")
+
+    build_command_for_codechecker = build_command.replace(read_in_file_path, analyze_dir_path + '/sources-root' + read_in_file_path)
+
+    codechecker_path = os.path.join(os.getcwd(), "build/CodeChecker/bin/CodeChecker")
```
Comment (gyorb) [inline remote/analyze_handler.py]: CodeChecker could be an argument to the script too.

File: remote/analyze_handler.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        scheduler.enter(10, 1, check_queue, ())
+        LOGGER.info("Check task queue in Redis.")
+        tasks = REDIS_DATABASE.lrange('ANALYSES_QUEUE', 0, -1)
+
+        task = REDIS_DATABASE.lpop('ANALYSES_QUEUE')
+
+        if task is not None:
+            LOGGER.info("Got a task: %s, starting analyze it, leftover tasks %s" % (task, len(tasks)))
+            analyze_id, part_number = str(task).split('-')
+            preAnalyze(analyze_id, part_number)
```
Comment (gyorb) [inline remote/analyze_handler.py]: I would handle every exception in `preAnalyze`, and `postAnalyze` so they are less likely to crash, however docker should restart the container in case of an error.

File: remote/analyze_handler.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python2
```
Comment (gyorb) [inline remote/analyze_handler.py:1]: Please add a short docstring about what the script does and for the main functions.

File: remote/analyze_handler.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        except Exception:
+            LOGGER.error("Failed to read in build command.")
+
+    file_to_analyze_path = os.path.join(analyze_dir_path, 'sources-root', 'file_path')
+
+    with open(file_to_analyze_path) as read_in_file:
+        try:
+            read_in_file_path = read_in_file.readline()
+        except Exception:
+            LOGGER.error("Failed to read in build command.")
```
Comment (gyorb) [inline remote/analyze_handler.py]: the same log message is here as before.

File: remote/analyze_handler.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+    file_path = os.path.join('/workspace', analyze_id)
+
+    source_file_without_extension = file_name + '_' + str(part_number)
+    source_file_with_extension = source_file_without_extension + file_extension
+
+    zip_file_path = os.path.join(file_path, source_file_with_extension)
+
+    analyze_dir_path = os.path.join(file_path, source_file_without_extension)
+
+    with zipfile.ZipFile(zip_file_path) as zf:
```
Comment (gyorb) [inline remote/analyze_handler.py:58]: Shouldn't this function return in case of some error. If extracting the files or getting the build command fails what can be done?
Maybe some extra analyze status values would be nice for these cases:
 - `preanalyze_failed`
- `analyze_failed` 
- `postanalyze_failed`

**Review by gyorb — CHANGES_REQUESTED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (tmsblgh) [reply docker-compose.yaml]: This is not needed, I have just forget to remove this.

File: remote/analyze_handler.py (hunk 2 above)
Comment (tmsblgh) [reply remote/analyze_handler.py]: Done

File: remote/analyze_handler.py (hunk 3 above)
Comment (tmsblgh) [reply remote/analyze_handler.py]: Done

File: remote/analyze_handler.py (hunk 6 above)
Comment (tmsblgh) [reply remote/analyze_handler.py]: Done

File: remote/analyze_handler.py (hunk 7 above)
Comment (tmsblgh) [reply remote/analyze_handler.py:58]: Good idea, will look into that, thanks!

File: remote/analyze_handler.py (hunk 5 above)
Comment (tmsblgh) [reply remote/analyze_handler.py:1]: Done

File: remote/analyze_handler.py (hunk 4 above)
Comment (tmsblgh) [reply remote/analyze_handler.py]: Done

**Review by gyorb — APPROVED** [review]
